src/main/java/ee/buerokratt/ruuter/domain/steps/http/HttpStep.java

```diff
@@ -54,14 +53,13 @@ private boolean isAllowedHttpStatusCode(ConfigurationInstance ci, ResponseEntity
     @Override
     public void handleFailedResult(ConfigurationInstance ci) {
         super.handleFailedResult(ci);
-        ApplicationProperties.DefaultAction defaultAction = ci.getProperties().getDefaultAction();
-        if (defaultAction != null && defaultAction.getService() != null) {
-            HttpQueryResponse response = ((HttpStepResult) ci.getContext().get(resultName)).getResponse();
-            HashMap<String, Object> body = defaultAction.getBody();
-            body.put("statusCode", response.getStatus().toString());
-            body.put("responseBody", ci.getMappingHelper().convertObjectToString(response.getBody()));
-            body.put("failedRequestId", MDC.get("spanId"));
-            ci.getConfigurationService().execute(defaultAction.getService(), defaultAction.getBody(), defaultAction.getQuery(), ci.getRequestOrigin());
+        if (!ci.getProperties().getHttpCodesAllowList().contains(((HttpStepResult) ci.getContext().get(resultName)).getResponse().getStatus())) {
+            HttpDefaultService propertiesHttpDefaultService = ci.getProperties().getDefaultServiceInCaseOfException();
+            if (httpDefaultService != null && httpDefaultService.getService() != null) {
```

Comment:
Maybe a better name for `httpDefaultService` would be `localHttpExceptionConfiguration` (and similarly for `propertiesHttpDefaultService ` => `globalHttpExceptionConfiguration` . They dont neccesarily have to be those names, its just that currenty it's a bit hard to understand what this code is supposed to do. I think better names for the variables would give a better idea to those unfamiliar with the business logic

Reply:
If we wanted to make the business logic even more clear, we could extract the conditons into separate methods:
* `httpDefaultService != null && httpDefaultService.getService() != null` => `localHttpExceptionConfExists()`
* `propertiesHttpDefaultService != null && propertiesHttpDefaultService.getService() != null` => `globalHttpExceptionConfExists()`

But maybe this is a bit overkill